wscript

```diff
@@ -69,52 +70,59 @@ def _setup_openmp(conf):
     else:
         conf.end_msg("OpenMP found.", 'GREEN')
 
+def _check_mac_osx_version(version):
+    '''
+     Native     Encoded
+    -------    --------
+     10.5.0 == 0x0A0500
+      ^ ^ ^       ^ ^ ^
+      | | |       | | |
+      major       major
+        | |         | |
+        minor       minor
+          |           |
+          patch       patch
+
+    version -> 24-bit encoded version
+    '''
+
+    assert isinstance(version, int)
+    floor_version = version
+    osx_version = 0
+
+    floor_version_major = (floor_version >> 8 & 0xff)
```

Comment:
Fancy but unreadable. Can anyone besides Joe debug or expand this in the future? If the answer is no, then I recommend something simpler. This script is only run once at build time, so performance is not an issue.